extensions/amp-form/0.1/amp-form.js

```diff
@@ -911,7 +911,14 @@ export class AmpForm {
     return response
       .json()
       .then(
-        json => this.handleSubmitSuccess_(/** @type {!JsonObject} */ (json)),
+        json => {
+          // Response.json() can return primitives e.g. bool, number.
+          const result =
+            typeof json === 'object'
+              ? /** @type {!JsonObject} */ (json)
+              : dict();
```

Comment:
This might be a behavior change. Currently if a truthy primitive is passed, I think it'll make it all the way through template render.